spec/lib/que/middleware/queue_collector_spec.rb

```diff
@@ -4,7 +4,7 @@
 
 RSpec.describe Que::Middleware::QueueCollector do
   subject(:collector) { described_class.new(->(_env) { nil }, options) }
-  let(:options) { {} }
+  let(:options) { {refresh_interval: 1.second} }
```

Comment:
This is being added because we don't have database cleaner and the state is being persisted across spec runs. Lowering the interval to 1.second forces the state to be refreshed between runs. 

@ankithads to change the spec to force a refresh before the flaky spec, and that is idependent of spec intervals. As a follow-up, we'll ensure that database state is cleared between examples (eg. using DatabaseCleaner).

Reply:
So the refresh time comes from the pg_stat_user_tables. Database cleaner does not help here.